hyprland: 0.43.0 -> 0.44.0, hyprpicker: 0.3.0 -> 0.4.1

fufexan opened the pull request:
Update hyprland and hyprpicker. More details in the commit messages.

[Hyprland CHANGELOG](https://github.com/hyprwm/Hyprland/releases/v0.44.0)
[Hyprpicker CHANGELOG](https://github.com/hyprwm/hyprpicker/releases/tag/v0.4.1)

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

NotAShelf commented on pkgs/by-name/hy/hyprland/package.nix:
```diff
@@ -45,30 +46,50 @@
+    fromJSON
+    readFile
+    ;
+  inherit (lib.asserts) assertMsg;
+  inherit (lib.attrsets) mapAttrsToList;
+  inherit (lib.lists)
+    flatten
+    concatLists
+    optionals
+    ;
+  inherit (lib.strings)
+    makeBinPath
+    optionalString
+    mesonBool
+    mesonEnable
+    ;
```
words can hardly express how much I hate this

fufexan replied:
The art of the glorious nixfmt.

NotAShelf reviewed (approved):
lgtm, love u fuf

Atemu commented:
Could you separate out the reformat part into a separate commit? It's impossible to tell what's going on otherwise.

JohnRTitor commented:
Seperate the mold linker commit as well.

fufexan committed:
hyprland: format

Use `inherit (lib) x` instead of `lib.x`
Remove no-longer-needed jq
Use importJSON

fufexan committed:
hyprland: use moldLinker via stdenvAdapters

Leave open the possibility of adding multiple adapters in the future.

fufexan committed:
hyprland: use meson for building

Follow upstream.

fufexan committed:
hyprland: 0.43.0 -> 0.44.0

Atemu commented:
I meant the hyprpicker commit.

It's fine to bundle changes and minor fixups related to an update into the update commit but automated reformats generally make it impossible to tell what other changes have actually taken place, so please separate that out into another commit.

Atemu commented on pkgs/by-name/hy/hyprland/package.nix:
```diff
@@ -45,30 +46,52 @@
+
+  # possibility to add more adapters in the future, such as keepDebugInfo,
+  # which would be controlled by the `debug` flag
+  adapters = flatten [
+    stdenvAdapters.useMoldLinker
+  ];
```
Nit: This strikes me as odd: Why do you flatten a list that only contains a function? This is a no-op.

fufexan replied:
Leftover from a previous attempt at also adding `(optional debug keepDebugInfo)`. I will remove it.

Atemu commented on pkgs/by-name/hy/hyprland/package.nix:
```diff
@@ -45,30 +46,52 @@
+    mesonEnable
+    ;
+
+  info = fromJSON (readFile ./info.json);
```
Nit: lib.importJSON

fufexan replied:
TIL

Atemu reviewed (approved):
A few nits, otherwise diff LGTM.

fufexan committed:
hyprpicker: format

fufexan committed:
hyprpicker: drop unused dependencies

fufexan committed:
hyprpicker: 0.3.0 -> 0.4.1

Atemu commented:
You forgot to update the info.json.

This would have been caught by a simple `Hyprland --version` check, so perhaps this would be a good time as any to introduce it.

Atemu commented:
https://github.com/NixOS/nixpkgs/pull/347624

fufexan commented:
> You forgot to update the info.json.

The problem is I forgot to update through the update script, and manually replaced the version and hash instead. Muscle memory.
